File: ruby/bubblesort/bubblesort.rb
# Write a function that will sort an array using Bubblesort in Ruby

def bubblesort(input_array)
  input_array.length.times do
    index = 0
    while index < input_array.length - 1
      if input_array[index] > input_array[index + 1]
        temp_element = input_array[index]
        input_array[index] = input_array[index + 1]
        input_array[index + 1] = temp_element
      end
      index += 1
    end
  end
  return input_array
end

p bubblesort([1, 0, -10, 10000, 993423, 234238423402384, 11, 11, 12, 12, 1239012309128312, 1223, 45065, 55, 344342, 234, 111])

File: javascript/udemy_data_and_algos/sorting/bubblesort/bubblesort.js
const numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0]

function bubbleSort(input_array) {
  for (let index_two = 0; index_two < input_array.length - 1; index_two += 1)  {
    for (let index = 0; index < input_array.length - 1; index += 1) {
      if (input_array[index] > input_array[index + 1]) {
        let temp = input_array[index]
        input_array[index] = input_array[index + 1]
        input_array[index + 1] = temp;
      }
    }
  }
  return input_array
}

bubbleSort(numbers)
console.log(numbers)
